### ISSUE TITLE
Got an error: Opening a multi-process MMKV instance with SINGLE_PROCESS_MODE!

### ISSUE THREAD

**Issue by samlu** [Tencent/MMKV#515]
Android mmkv 1.2.2

This may be related to issue #513 

From the crashlytics report, I can see the following error:
**IllegalArgumentException: Opening a multi-process MMKV instance [loga] with SINGLE_PROCESS_MODE!**

I am sure that this "loga" mmkv is only used in one process so it was opened with the SINGLE_PROCESS_MODE.

and the mmkv log is:
**ERR MMKV_Android.cpp:237::checkProcessMode: Fail to share lock [loga]**

Again, we cannot reproduce this problem only our device and it only happens on a few Android devices.

**Comment by samlu** [comment]
Here is the complete mmkv log:
```
INFO MMKV_Android.cpp:158::mmkvWithID: prepare to load loga (id 5040b950234780a5be12a91ed3f8430f) from rootPath /mnt/expand/3d5e19fa-3650-4842-a2af-ee350219d858/user/0/<pkg_name>/app_log
ERR MemoryFile.cpp:139::reloadFromFile: fail to open:/mnt/expand/3d5e19fa-3650-4842-a2af-ee350219d858/user/0/<pkg_name>/app_log/5040b950234780a5be12a91ed3f8430f, Invalid argument
ERR MemoryFile.cpp:139::reloadFromFile: fail to open:/mnt/expand/3d5e19fa-3650-4842-a2af-ee350219d858/user/0/<pkg_name>/app_log/5040b950234780a5be12a91ed3f8430f.crc, Invalid argument
ERR MemoryFile.cpp:139::reloadFromFile: fail to open:/mnt/expand/3d5e19fa-3650-4842-a2af-ee350219d858/user/0/<pkg_name>/app_log/5040b950234780a5be12a91ed3f8430f, Invalid argument
ERR MMKV_IO.cpp:73::loadFromFile: file [/mnt/expand/3d5e19fa-3650-4842-a2af-ee350219d858/user/0/<pkg_name>/app_log/5040b950234780a5be12a91ed3f8430f] not valid
ERR MMKV_Android.cpp:237::checkProcessMode: Fail to share lock [5040b950234780a5be12a91ed3f8430f]
```

It looks like mmkv opens the mmkv file with an unknown error at the first beginning.
For this case, throw an "Opening a multi-process MMKV instance with SINGLE_PROCESS_MODE!" exception may mislead developers.

**Comment by lingol** [comment]
I see.

**Comment by lingol** [comment]
Fixed in the dev branch.

**Comment by m30102** [comment]
同样的问题，首次启动在没有获取权限的情况下调用mmkv会报这个错。<MemoryFile.cpp:139::reloadFromFile> fail to open:/storage/emulated/0/ccb/smartcanteen/mmkv//mmkv.default, Permission denied

**Comment by lingol** [comment]
> 同样的问题，首次启动在没有获取权限的情况下调用mmkv会报这个错。MemoryFile.cpp:139::reloadFromFile fail to open:/storage/emulated/0/ccb/smartcanteen/mmkv//mmkv.default, Permission denied

No. I don't think that an issue that should be solved by MMKV. Solve it by your own App bussiness logic.

**Comment by lingol** [comment]
Released with [v1.2.3](https://github.com/Tencent/MMKV/releases/tag/v1.2.3).
